File: src/main/kotlin/limia/Exception/EntityAlreadyExistsException.java
package limia.Exception;

import java.util.ArrayList;

/**
 * Created by workstation on 16/04/2017.
 */
public class EntityAlreadyExistsException extends Exception{


    private ArrayList<Class> types;

    public ArrayList<Class> getTypes() {
        return types;
    }

    public void setTypes(ArrayList<Class> types) {
        this.types = types;
    }

    public void addEntityType(Class type) {
        types.add(type);
    }

    public EntityAlreadyExistsException() {
    }

    public EntityAlreadyExistsException(String message) {
        super(message);
    }

    public EntityAlreadyExistsException(String message, Throwable cause) {
        super(message, cause);
    }

    public EntityAlreadyExistsException(Throwable cause) {
        super(cause);
    }
}

File: src/main/kotlin/limia/Exception/EntityAlreadyExistsException.kt
package limia.Exception

import java.util.ArrayList

/**
 * Created by workstation on 16/04/2017.
 */
class EntityAlreadyExistsException : Exception {


    var types: ArrayList<Class<*>>? = null

    fun addEntityType(type: Class<*>) {
        types!!.add(type)
    }

    constructor() {}

    constructor(message: String) : super(message) {}

    constructor(message: String, cause: Throwable) : super(message, cause) {}

    constructor(cause: Throwable) : super(cause) {}
}
